packages/core/src/baseCogniteClient.ts

```diff
@@ -441,6 +441,45 @@ export default class BaseCogniteClient {
     }
   }
 
+  /**
+   * Returns IDP id token
+   *
+   * ```js
+   * client.loginWithOAuth({ cluster: 'bluefield', ... });
+   * await client.authenticate();
+   * const idToken = await client.getIdToken();
+   * ```
+   */
+  public async getIdToken(): Promise<string | null> {
+    switch (this.flow!.type) {
+      case 'AAD_OAUTH': {
+        if (this.azureAdClient) {
+          return this.azureAdClient.getAccountToken();
+        } else {
+          return Promise.resolve(null);
+        }
+      }
+      case 'ADFS_OAUTH': {
+        if (this.adfsClient) {
```

Comment:
Maybe cleaner with: 
`return this.adfsClient ? this.adfsClient.getIdToken() : null;`
?